library/swift/src/RetryPolicy.swift

```diff
@@ -47,13 +47,15 @@ public final class RetryPolicy: NSObject {
   ///                                     retries. Spans the point at which the entire downstream
   ///                                     request has been processed and when the upstream
   ///                                     response has been completely processed.
-  ///                                     Zero may be specified to disable the upstream timeout.
+  ///                                     Nil may be specified to disable the upstream timeout.
   public init(maxRetryCount: UInt, retryOn: [RetryRule], perRetryTimeoutMS: UInt? = nil,
-              totalUpstreamTimeoutMS: UInt = 15_000)
+              totalUpstreamTimeoutMS: UInt? = 15_000)
   {
-    if let perRetryTimeoutMS = perRetryTimeoutMS {
-      assert(perRetryTimeoutMS <= totalUpstreamTimeoutMS || totalUpstreamTimeoutMS == 0,
-             "Per-retry timeout must be <= total timeout if total timeout != 0")
+    if let perRetryTimeoutMS = perRetryTimeoutMS,
+      let totalUpstreamTimeoutMS = totalUpstreamTimeoutMS
+    {
+      assert(perRetryTimeoutMS <= totalUpstreamTimeoutMS,
```

Comment:
you still need to check for `totalUpstreamTimeoutMS ` here, right?

Reply:
assert(perRetryTimeoutMS <= totalUpstreamTimeoutMS && totalUpstreamTimeoutMS != 0)

Reply:
Personally I think that omitting the `totalUpstreamTimeoutMS != 0` exclusion from the assert is probably preferred here since the consumer will then change the `0` to `nil` to achieve the same behavior (in line with the documentation of this function). Thoughts?

Reply:
One question - What happens if I provide `perRetryTimeoutMS = 0` and `totalUpstreamTimeoutMS = 0`? I would expect this too break since the docs says that a user should use `nil` if they want `totalUpstreamTimeoutMS` to be ignored.

Reply:
That's fair, I can update to include it in the assert.